minidisc-rs/src/netmd/encryption.rs

```diff
@@ -0,0 +1,67 @@
+use std::thread;
+use cbc::cipher::block_padding::NoPadding;
+use cbc::cipher::{KeyInit, BlockDecryptMut, KeyIvInit, BlockEncryptMut};
+use tokio::sync::mpsc::{UnboundedReceiver, unbounded_channel};
+use rand::RngCore;
+
+use super::interface::DataEncryptorInput;
+
+type DesEcbEnc = ecb::Decryptor<des::Des>;
+type DesCbcEnc = cbc::Encryptor<des::Des>;
+
+pub fn new_thread_encryptor(_input: DataEncryptorInput) -> UnboundedReceiver<(Vec<u8>, Vec<u8>, Vec<u8>)> {
+    let (tx, rx) = unbounded_channel::<(Vec<u8>, Vec<u8>, Vec<u8>)>();
+    let input = Box::from(_input);
+    thread::spawn(move || {
+        let mut iv = [0u8; 8];
+
+        // Create the random key
+        let mut random_key = [0u8; 8];
+        rand::thread_rng().fill_bytes(&mut random_key);
+
+        // Encrypt it with the kek
+        let mut encrypted_random_key = random_key.clone();
+        match DesEcbEnc::new(&input.kek.into()).decrypt_padded_mut::<NoPadding>(&mut encrypted_random_key){
+            Err(x) => panic!("Cannot create main key {:?}", x),
```

Comment:
Panics should be avoided here, this should just return an error.

Since this crate is intended to be used as a library, the effect of a `panic!()` is that a higher level crate using this one has no way to handle the error at all, and it would just crash. Even if this is fatal, that should be passed on through a return type.